### PR TITLE
fix: #2155 DeepSeek reasoning_content missing in tool call messages

### PR DESCRIPTION
**Pull request by wahmd**
This PR fixes issue #2155 (400 error on tool calls due to missing `reasoning_content` in reconstructed assistant messages.)

**Root Cause**: When DeepSeek returns a response with `reasoning_content` and `tool_calls`, the SDK converts it to:
- A `ResponseReasoningItem` (stores reasoning in `summary` field)
- A `ResponseFunctionToolCall` (stores tool call)

On the next API call, `Converter.items_to_messages()` reconstructs the assistant message with tool calls, but was only handling reasoning content for Anthropic/Claude models (via thinking blocks), not DeepSeek.

**Solution**: The fix follows the same pattern as Anthropic's thinking blocks but uses DeepSeek's `reasoning_content` field instead. It:
1. Extracts reasoning text from the `summary` field when processing reasoning items for DeepSeek models
2. Stores it temporarily in `pending_reasoning_content`
3. Adds it to the assistant message when tool calls are processed

### Checks

- [x] I've added new tests 
- [x] I've run `make lint` and `make format` (linter shows only import resolution warnings, not code issues)
- [x] I've made sure tests pass

Note: The fix works for both code paths:
- **LiteLLM path**: When using `LitellmModel` with `model="deepseek/deepseek-reasoner"`
- **OpenAI ChatCompletions path**: When using `OpenAIChatCompletionsModel` with `model="deepseek-reasoner"` and DeepSeek's OpenAI-compatible endpoint

### CODE REVIEW

File: src/agents/models/chatcmpl_converter.py [hunk 1]
```diff
@@ -579,6 +580,11 @@ def ensure_assistant_message() -> ChatCompletionAssistantMessageParam:
             elif func_call := cls.maybe_function_tool_call(item):
                 asst = ensure_assistant_message()
 
+                # If we have pending reasoning content for DeepSeek, add it to the assistant message
+                if pending_reasoning_content:
+                    asst["reasoning_content"] = pending_reasoning_content  # type: ignore[typeddict-unknown-key]
+                    pending_reasoning_content = None  # Clear after using
```
[the remark this answers is not in the document]
Comment (wahmd) [reply src/agents/models/chatcmpl_converter.py:590]: good point, cleared in `flush_assistant_message`. 
flush_assistant_message() runs before processing new messages, so stale `pending_reasoning_content` is cleared preventing contamination.

Also, clears it when there's no current assistant message (edge case)

**Comment by seratch** [comment]
@codex review again

**Comment by seratch** [comment]
can you fix the lint issue?

File: tests/test_agent_runner_sync.py [hunk 2]
```diff
@@ -9,9 +9,9 @@
 
 
 @pytest.fixture
-def fresh_event_loop_policy() -> Generator[asyncio.AbstractEventLoopPolicy, None, None]:
```
Comment (wahmd) [inline tests/test_agent_runner_sync.py:12]: just FYI, changed this because `make mypy` was failing. 
asyncio.AbstractEventLoopPolicy" is not defined_

Comment (wahmd) [reply tests/test_agent_runner_sync.py:12]: Also, lint and format issues are now resolved. Marking "ready for review".

File: src/agents/models/chatcmpl_converter.py [hunk 3]
```diff
@@ -687,6 +697,17 @@ def ensure_assistant_message() -> ChatCompletionAssistantMessageParam:
                     # This preserves the original behavior
                     pending_thinking_blocks = reconstructed_thinking_blocks
 
+                # DeepSeek requires reasoning_content field in assistant messages with tool calls
+                elif model and "deepseek" in model.lower():
+                    summary_items = reasoning_item.get("summary", [])
```
[the remark this answers is not in the document]
Comment (wahmd) [reply src/agents/models/chatcmpl_converter.py:711]: resolved.

**Comment by seratch** [comment]
can you fix the mypy error?

**Review by seratch — APPROVED** [review]

**Comment by Sourav-Nandy-ai** [comment]
Hi Waleed,

Your fix for DeepSeek's missing reasoning_content field on tool-call messages is the kind of correctness fix where confirming "the crash is gone" and confirming "tool-call behavior actually stabilized across a distribution of runs" are two different questions. The first is easy to eyeball, while the second usually is not.

Since you would find out anyway if you tried it, I want to mention that when I first looked at this PR, the OpenAI Agents SDK integration for agent-eval (agent_regress) would have crashed on the very first call against any agent. It was calling a method that does not exist on the SDK's Agent class.

That is fixed now. It calls the SDK's actual Runner.run() entrypoint, and I independently reproduced this end-to-end against a live openai-agents install (with 6 out of 6 integration tests passing) rather than just a unit-test mock.

I am not claiming I have run agent-eval against your DeepSeek and tool-call scenario specifically, as I have not. However, the underlying idea (running an agent N times pre-fix and N times post-fix to get a Mann-Whitney p-value, Cohen's d, and bootstrap CI instead of simply eyeballing whether the 400 error is gone) is built exactly for provider-specific fixes like this one.

Given you are building EMR tooling where a silent tool-call behavior shift would actually matter, I would be interested in your take on whether this is useful now that the integration itself is not broken on contact.

You can find the repository here: https://github.com/RudrenduPaul/agent-eval

Best,
Sourav
